src/user/models/kyc/kyc.service.ts

```diff
@@ -175,13 +183,44 @@ export class KycService {
     const dataComplete = this.isDataComplete(user);
     if (!dataComplete) throw new BadRequestException('Ident data incomplete');
 
+    const users = await this.userDataService.getUsersByInformation(user);
+    if (users) {
+      const completedUser = users.find((data) => KycCompleted(data.kycStatus));
+      const shouldSendLinkEmail = users.length > 1;
+      if (shouldSendLinkEmail && completedUser) {
+        const oldestToNewestUser = completedUser.users.sort((a, b) => (a.created > b.created ? 1 : -1));
+        const existingAddress = oldestToNewestUser[0].address;
+        const newAddress = user.users[0].address;
+
+        const linkAddress = await this.linkAddressRepo.save(LinkAddress.create(existingAddress, newAddress));
+
+        await this.mailService.sendTranslatedMail({
+          userData: user,
+          translationKey: 'mail.link.address',
+          params: {
+            firstname: completedUser.firstname,
+            surname: completedUser.surname,
+            organizationName: completedUser.organizationName ?? '',
+            existingAddress: Blank(existingAddress, BlankType.WALLET_ADDRESS),
+            newAddress: Blank(newAddress, BlankType.WALLET_ADDRESS),
+            url: this.buildLinkUrl(linkAddress.authentication),
+          },
+        });
+        throw new ConflictException('User already has completed Kyc');
```

Comment:
Is this error handled on payment page?

Reply:
on the new version yes, this is mandatory to show the dialog, that an email was sent

Reply:
Perfect 👍